src/zcbor_decode.c

```diff
@@ -404,24 +404,30 @@ static bool str_start_decode(zcbor_state_t *state,
 	return true;
 }
 
+
+static bool str_overflow_check(zcbor_state_t *state, size_t len)
+{
+	/* Casting to size_t is safe since value_extract() checks that
+	 * payload_end is bigger that payload. */
+	if (len > (size_t)(state->payload_end - state->payload)) {
+		zcbor_log("error: 0x%zu > 0x%zu\r\n",
+			len,
+			(state->payload_end - state->payload));
+		ERR_RESTORE(ZCBOR_ERR_NO_PAYLOAD);
```

Comment:
Does this return? If so, doesn't it violate the "macros should not alter code flow" rule we have in ncs/zephyr?